packages/hub/services/in-app-purchases.ts

```diff
@@ -1,8 +1,56 @@
+import config from 'config';
+import fetch from 'node-fetch';
+
+interface InAppPurchaseValidationResult {
+  valid: boolean;
+  response: any;
+}
+
 export default class InAppPurchases {
-  async validate(provider: string, receipt: any) {
-    console.error('In-app purchase validation is not implemented, arguments are', provider, receipt);
+  async validate(provider: string, receipt: any): Promise<InAppPurchaseValidationResult> {
+    if (provider === 'apple') {
+      return this.validateFromApple(receipt);
+    }
+
+    if (provider === 'google') {
+      return this.validateFromGoogle(receipt);
+    }
+
+    throw new Error(`In-app purchase validation is not implemented for provider ${provider}; receipt: ${receipt}`);
+  }
+
+  private async validateFromApple(receipt: string): Promise<InAppPurchaseValidationResult> {
+    let requestBody = {
+      'receipt-data': receipt,
+    };
+
+    let response = await fetch(config.get('iap.apple.verificationUrl'), {
+      method: 'POST',
+      body: JSON.stringify(requestBody),
+    });
+
+    let json = await response.json();
+
+    if (json.status === 0) {
+      return { valid: true, response: json };
+    } else {
+      return { valid: false, response: json };
```

Comment:
There are some cases where we want to report to Sentry listed here:

https://developer.apple.com/documentation/appstorereceipts/status

Most things besides 21003 seem like they should be reported to Sentry?